grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -114,6 +114,10 @@ class LocationService {
         def locations = new HashSet()
         locations += getLocations(fields, params)
 
+        if (params.locationChooser) {
```

Comment:
I don't love this. A more generic way would be better. Like adding a backwards compatible boolean flag to the method. We don't need to change any of the existing calls to this method, but we can override the one we need to. 
```
def getLocations(String[] fields, Map params, Boolean isSuperuser, String direction, Location currentLocation, User user, Boolean excludeDisabled = Boolean.FALSE) {
    def locations = new HashSet()
    locations += getLocations(fields, params)
    if (excludeDisabled) { 
        locations = locations.findAll { Location location -> location.status == LocationStatus.ENABLED 
    }
}
```
@awalkowiak do you have any thoughts?

Reply:
Yeah, I think I like your version more. In the current version, we are making decisions about what we want to pull based on the `params.locationChooser` in the controller. @kchelstowski let's go with this version for now.

Btw @jmiranda there is already a tech debt ticket created for refactoring the location API (OBPIH-4594), perhaps we could squeeze it in the 0.8.20.